python/nano/src/bigdl/nano/pytorch/trainer/Trainer.py

```diff
@@ -156,3 +156,79 @@ def compile(model: nn.Module,
+                 metric: Metric = None,
+                 backend='inc',
+                 conf='',
+                 framework='pytorch_fx',
+                 approach='static',
+                 tuning_strategy='bayesian',
+                 accuracy_criterion: dict = None,
+                 timeout=0,
+                 max_trials=1
+                 ) -> Optional[GraphModule]:
+        """
+        Calibrate a Pytorch-Lightning model for post-training quantization.
+
+        :param pl_model:       A Pytorch-Lightning model to be quantized.
+        :param calib_dataloader:    Iterable dataloader for calibration. Required for static
+                                    quantization.
+        :param val_dataloader:      Iterable dataloader for evaluation.
+        :param metric:              Metric for evaluation.
+        :param backend:             inc or nncf(nncf is not supported yet). Default: inc
+        :param conf:        A path to conf yaml file for quantization.
+                            Default: None, use default config.
+        :param framework:   Supported values are pytorch, pytorch_fx, pytorch_ipex; allow new
+                            framework backend extension. Default: pytorch_fx.
+                            Consistent with Intel Neural Compressor Quantization.
+        :param approach:    static or dynamic.
+                            static: post_training_static_quant,
+                            dynamic: post_training_dynamic_quant,
+                            Default: static.
+        :param tuning_strategy:    bayesian, basic, mse, sigopt. Default: bayesian.
+        :param accuracy_criterion:  Tolerable accuracy drop.
+                                    accuracy_criterion = {'relative': 0.1, higher_is_better=True}
+                                    allows relative accuracy loss: 1%. accuracy_criterion =
+                                    {'absolute': 0.99, higher_is_better=False} means accuracy < 0.99
+                                     must be satisfied.
+        :param timeout:     Tuning timeout (seconds). Default: 0,  which means early stop.
+                            Combine with max_trials field to decide when to exit.
+        :param max_trials:  Max tune times. Default: 1.
+                            Combine with timeout field to decide when to exit.
+                            "timeout=0, max_trials=1" means it will try quantization only once and
+                            return satisfying best model.
+        :return:             A GraphModule. If there is no model found, return None.:
+        """
+        if backend == 'inc':
+            from bigdl.nano.quantization import QuantizationINC
+            from neural_compressor.experimental import common
+            approach_map = {
+                'static': 'post_training_static_quant',
+                'dynamic': 'post_training_dynamic_quant'
+            }
+            approach = approach_map.get(approach)
```

Comment:
need to check `approach` is valid

Reply:
Add following:
```python
            if approach not in ['static', 'dynamic']:
                raise ValueError("Approach should be 'static' or 'dynamic', "
                                 "{} is invalid.".format(approach))
```
And added an extra test case to test if error is raised as expected.